### PR TITLE
refactor: share options to include squad

### PR DESCRIPTION
**Pull request by sshanzel**
### Describe what this PR does
- Refactored the `ProfileLink` component to ask for the href only instead of the whole user object which only needs the permalink.
- Conditions when to the squad options:
  - When it is not a comment (as the user's intent is to share the comment and shared items in Squad are posts only.
  - When the post is public.

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-977 #done

### CODE REVIEW

File: packages/extension/src/companion/useRawBackgroundRequest.ts [hunk 1]
```diff
@@ -1,8 +1,9 @@
+import { EmptyObjectLiteral } from '@dailydotdev/shared/src/lib/kratos';
 import { useEffect } from 'react';
 import { browser } from 'webextension-polyfill-ts';
 
 export const useRawBackgroundRequest = (
-  command: (params: unknown) => void,
+  command: (params: EmptyObjectLiteral) => void,
```
Comment (sshanzel) [inline packages/extension/src/companion/useRawBackgroundRequest.ts:6]: I just noticed an error while having the extension projects opened.

**Review by rebelchris — CHANGES_REQUESTED** [review]
No remarks, but wait with merging please as there is an ongoing discussion here:
https://dailydotdev.slack.com/archives/C03QZBF46JC/p1675365375994899

I'll request changes for now, until we get a resolution sorry for that.

**Comment by sshanzel** [comment]
No worries, we can wait til decision is made 🚢

**Review by rebelchris — APPROVED** [review]
Alright good to go 🙏

File: packages/shared/src/components/widgets/SocialShare.tsx [hunk 2]
```diff
@@ -102,6 +118,19 @@ export const SocialShare = ({
         onClick={() => trackClick(ShareProvider.Email)}
         label="Email"
       />
+      {!isComment &&
```
Comment (el) [inline packages/shared/src/components/widgets/SocialShare.tsx:121]: There is an alignment issue
SCR-20230203-of5

File: packages/shared/src/components/widgets/SocialShare.tsx [hunk 3]
```diff
@@ -102,6 +118,19 @@ export const SocialShare = ({
...
+        !post.private &&
+        squads.map((squad) => (
+          <button
+            type="button"
+            className="flex flex-col items-center"
+            key={squad.id}
+            onClick={() => onShareToSquad(squad)}
+          >
+            <SourceProfilePicture source={squad} />
+            <ShareText className="mt-2">@{squad.handle}</ShareText>
```
Comment (el) [inline packages/shared/src/components/widgets/SocialShare.tsx:131]: Shouldnt we use the name?

**Review by el — CHANGES_REQUESTED** [review]

File: packages/shared/src/components/widgets/SocialShare.tsx (hunk 3 above)
Comment (sshanzel) [reply packages/shared/src/components/widgets/SocialShare.tsx:131]: Names are often so much longer and can be the same with other Squads. I think going with the handle has its benefits. WDYT?

File: packages/shared/src/components/widgets/SocialShare.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/shared/src/components/widgets/SocialShare.tsx:121]: Oh, good catch! Let me check this quickly! 🚢

Comment (sshanzel) [reply packages/shared/src/components/widgets/SocialShare.tsx:121]: Context behind the solution: the WhatsApp icon has the correct size but the label below made its total width expand, making it unequal to the email share, which led to making it look like the Squads weren't in the same alignment. I turned it into a grid to make them have the same total width sizes across the board.

Normal size:
image

Mobile:
image

Where there are long labels:
image

Let me know what you think!

**Comment by sshanzel** [comment]
I have also refactored the order based on the discussion in Slack.
https://dailydotdev.slack.com/archives/C03QZBF46JC/p1675571154715349?thread_ts=1675525512.721859&cid=C03QZBF46JC

File: packages/shared/src/components/widgets/SocialShare.tsx (hunk 2 above)
Comment (el) [reply packages/shared/src/components/widgets/SocialShare.tsx]: Yeap, looks great, nice work. Just make sure it also works well on narrower phones.

**Review by el — APPROVED** [review]
Great work ❤️

**Pull request merged**